src/test/scala/org/broadinstitute/dsde/rawls/expressions/SlickSimpleExpressionParserTest.scala

```diff
@@ -114,169 +201,162 @@ class SlickSimpleExpressionParserTest extends FunSuite with TestDriverComponent
       }
 
-      assert {
-        evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.").isFailure
+      intercept[RawlsException] {
+        runAndWait(evalFinalAttribute(workspaceContext, "Sample", "sample1", "workspace."))
       }
 
 // https://broadinstitute.atlassian.net/browse/GAWB-349 figure out how to add this back in one day
 //      assert {
-//        evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.missing.also_missing").isFailure
+//        evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.missing.also_missing").isFailure
 //      }
 
-      assert {
+      intercept[RawlsException] {
         val attributesPlusReference = testData.workspace.attributes + ("sample1ref" -> AttributeEntityReference("Sample", "sample1"))
         runAndWait(workspaceQuery.save(testData.workspace.copy(attributes = attributesPlusReference)))
 
-        evaluator.evalFinalAttribute(workspaceContext, "dummy text", "dummy text", "workspace.sample1ref.").isFailure
+        runAndWait(evalFinalAttribute(workspaceContext, "Sample", "sample1", "workspace.sample1ref."))
       }
     }
   }
 
   test("simple entity expression") {
     withTestWorkspace { workspaceContext =>
-      val evaluator = expressionEvaluator
 
-      assertResult(Seq(testData.sample2)) {
-        runAndWait(evaluator.evalFinalEntity(workspaceContext, "Pair", "pair1", "this.case").get)
+      assertResult(Set("sample2")) {
+        runAndWait(evalFinalEntity(workspaceContext, "Pair", "pair1", "this.case")).map(_.name).toSet
       }
 
-      assertResult(Seq(testData.sample1, testData.sample2, testData.sample3)) {
-        runAndWait(evaluator.evalFinalEntity(workspaceContext, "Individual", "indiv1", "this.sset.samples").get)
+      assertResult(Set("sample1", "sample2", "sample3")) {
+        runAndWait(evalFinalEntity(workspaceContext, "Individual", "indiv1", "this.sset.samples")).map(_.name).toSet
       }
 
-      assertResult(Seq(testData.pair1)) {
-        runAndWait(evaluator.evalFinalEntity(workspaceContext, "Pair", "pair1", "this").get)
+      assertResult(Set("pair1")) {
+        runAndWait(evalFinalEntity(workspaceContext, "Pair", "pair1", "this")).map(_.name).toSet
       }
     }
   }
 
   test("workspace entity expression") {
     withTestWorkspace { workspaceContext =>
-      val evaluator = expressionEvaluator
 
-      assertResult(Seq(testData.sample1)) {
+      assertResult(Set("sample1")) {
         val attributesPlusReference = testData.workspace.attributes + ("sample1ref" -> AttributeEntityReference("Sample", "sample1"))
         runAndWait(workspaceQuery.save(testData.workspace.copy(attributes = attributesPlusReference)))
 
-        runAndWait(evaluator.evalFinalEntity(workspaceContext, "dummy text", "dummy text", "workspace.sample1ref").get)
+        runAndWait(evalFinalEntity(workspaceContext, "Pair", "pair1", "workspace.sample1ref")).map(_.name).toSet
```

Comment:
We used to evaluate workspace expressions regardless of the root entity, because we never had to go to look at the root entity in order to evaluate the expression. This seemed like an odd use case and Doge agreed it didn't make much sense so I didn't preserve it.